planner/core/logical_plan_builder.go

```diff
@@ -1278,6 +1393,31 @@ func (b *PlanBuilder) buildProjectionFieldNameFromExpressions(_ context.Context,
 	}
 }
 
+func buildExpandFieldName(expr expression.Expression, name *types.FieldName, genName string) *types.FieldName {
+	_, isCol := expr.(*expression.Column)
+	var origTblName, origColName, dbName, colName, tblName model.CIStr
+	if genName != "" {
+		// for case like: gid_, gpos_
+		colName = model.NewCIStr(expr.String())
+	} else if isCol {
+		// col ref to original col, while its nullability may be changed.
```

Comment:
just a gen_name to avoid using the same name as the original col did.
for example, we have a case:
select count(a) from t group by a, b with rollup.
since the final agg asks for source column a, and we have the schema like a, a, b before constructing the final agg.
when resolving the arg in count(?), there are something happens:
```
(er *expressionRewriter) toColumn(v *ast.ColumnName) {
// first according to their names to resolve ast column name.
idx, err := expression.FindFieldName(er.names, v)
```

here first according to their names to resolve an ast-column name. if a and a' have the same name, there will be a duplicated name error.

ex_ prefix is adequate, since this internal col is not visible to user, it's only used by our internal computation

Reply:
Which attribute name is the last displayed to the user?